opslevel/datasource_opslevel_users_all.go

```diff
@@ -55,19 +61,26 @@ func (d *UserDataSourcesAll) Schema(ctx context.Context, req datasource.SchemaRe
 
 func (d *UserDataSourcesAll) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
 	var planModel, stateModel userDataSourcesAllModel
+	var users *opslevel.UserConnection
+	var err error
 
 	resp.Diagnostics.Append(req.Config.Get(ctx, &planModel)...)
-
 	if resp.Diagnostics.HasError() {
 		return
 	}
 
-	users, err := d.client.ListUsers(nil)
+	if planModel.IgnoreDeactivated.ValueBool() {
+		withoutDeactivedUsers := d.client.InitialPageVariablesPointer().WithoutDeactivedUsers()
+		users, err = d.client.ListUsers(withoutDeactivedUsers)
+	} else {
+		users, err = d.client.ListUsers(nil)
+	}
```

Comment:
The beauty of the builder pattern is you can write logic like this which is slightly simpler to read

```suggestion
	vars := d.client.InitialPageVariablesPointer()
	if planModel.IgnoreDeactivated.ValueBool() {
		vars.WithoutDeactivedUsers()
	}
	users, err = d.client.ListUsers(vars)
```

Reply:
This suggestion won't work because ListUsers() needs `nil` or `vars` containing `"after":`, `"first":` and `filter:`
 
`vars := d.client.InitialPageVariablesPointer()` only sets `"after":` and `"first":` which will cause the query to fail

Reply:
is there a reason we didn't put `filter == null` into the default vars?

Reply:
Adding `filter = null` to the defaults causes lots of errors